qiskit_nature/deprecation.py

```diff
@@ -228,6 +228,61 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def deprecate_positional_arguments(
+    version: str,
+    func_name: str,
+    pre_positional_kw: List[str],
+    additional_msg: Optional[str] = None,
+    stack_level: int = 3,
+) -> Callable:
+    """Decorator to convert positional arguments into keyword arguments and warn upon use.
+    .. code-block:: python
+
+        @deprecate_positional_arguments("0.1", (b, d))
+        def function(a, b, c, d, **kwargs):
+            returns a + c + sum(kwargs.values())
+
+        # The following two calls would be equivalent
+        function(1, 2, 3, 4)  # Would raise deprecation warning for arguments b and d.
+        function(a=1, b=2, c=3, d=4)  # Would not raise any warning.
+
+    Args:
+        version: Version to be used
+        func_name: Name of the function where the deprecation takes place will be used to
+        write the deprecation message.
+        pre_positional_kw: List of arguments of the function before the deprecation.
```

Comment:
If `pre_positional_kw` is the list of positional arguments prior to the deprecation, how do you actually specify which arguments are to be deprecated?

Reply:
I access the name of each keyword argument and the ones that are in one list and not the other are the ones to be deprecated.
So, I iterate over pre_positional_kw, and for each one I check if that keyword is in `current_positional_kw = inspect.getfullargspec(func)[0]`. Then I raise a message if it isn't and in both cases add it to kwargs dictionary with I use to call the decorated funtion in the end.

Reply:
So would it then not make more sense to call this the `deprecated_positiional_kw`?
And if so, why does one of the cases I saw in your PR include `self` in this list? Surely this should not be deprecated

Reply:
Right, thanks for the detailed explanation :+1: 
In that case I would suggets:
- calling this argument: `old_function_arguments` or `original_function_arguments`
- expanding the explanation in this docstring to match your explanation here

Reply:
Done